presto-main/src/main/java/io/prestosql/sql/planner/optimizations/CheckSubqueryNodesAreRewritten.java

```diff
@@ -50,12 +52,23 @@ public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, Sym
                 .findFirst()
                 .ifPresent(node -> {
                     CorrelatedJoinNode correlatedJoinNode = (CorrelatedJoinNode) node;
-                    throw error(correlatedJoinNode.getCorrelation(), correlatedJoinNode.getOriginSubquery());
+                    throw error(correlatedJoinNode);
                 });
 
         return plan;
     }
 
+    private PrestoException error(CorrelatedJoinNode correlatedJoinNode)
+    {
+        // FULL JOIN LATERAL with condition other than TRUE is not supported neither with non-empty, nor with empty correlation.
+        if (correlatedJoinNode.getType() == FULL
+                && !correlatedJoinNode.getFilter().equals(TRUE_LITERAL)
+                && correlatedJoinNode.getCorrelation().isEmpty()) {
+            throw semanticException(NOT_SUPPORTED, correlatedJoinNode.getOriginSubquery(), "FULL JOIN involving LATERAL relation is not supported with condition other than TRUE");
```

Comment:
This should be a check in the analyzer, not in the optimizer.

Reply:
The Optimizer could simplify the filter expression to TRUE. If we do the check in the Analyzer, we lose the chance. Maybe we could let `... FULL JOIN LATERAL ... ON xxx` pass the Analyzer and let it eventually fail here with "Given correlated subquery is not supported" (instead of inventing a new error message)?

Reply:
Yes, that's better, but if we could detect some cases earlier during analysis and produce a better message that'd be ideal. The only reason we don't do it generically is that it's hard to tell whether a particular shape of correlated query is supported at that time.

Reply:
So what's the conclusion? Should it be a check in the Analyzer with an explicit message that `... FULL JOIN LATERAL ...` must go with `TRUE` condition?

Reply:
Let's stick with what you suggested for now and improve later if necessary.